core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -1029,34 +1021,22 @@ fn do_cast_string_to_int<
     type_name: &str,
     min_value: T,
 ) -> CometResult<Option<T>> {
-    let len = str.len();
-    if str.is_empty() {
+    let trimmed_str = str.trim();
+    if trimmed_str.is_empty() {
         return none_or_err(eval_mode, type_name, str);
     }
-
+    let len = trimmed_str.len();
     let mut result: T = T::zero();
     let mut negative = false;
     let radix = T::from(10);
     let stop_value = min_value / radix;
-    let mut state = State::SkipLeadingWhiteSpace;
-    let mut parsed_sign = false;
-
-    for (i, ch) in str.char_indices() {
-        // skip leading whitespace
-        if state == State::SkipLeadingWhiteSpace {
-            if ch.is_whitespace() {
-                // consume this char
-                continue;
-            }
-            // change state and fall through to next section
-            state = State::ParseSignAndDigits;
-        }
+    let mut parse_sign_and_digits = true;
 
-        if state == State::ParseSignAndDigits {
-            if !parsed_sign {
+    for (i, ch) in trimmed_str.char_indices() {
```

Comment:
We process the trimmed string here and we no longer need the logic for skipping leading and trailing whitespace